test-utils/src/lib.rs

```diff
@@ -1,16 +1,60 @@
-pub use miden::{MemAdviceProvider, ProgramInfo, ProofOptions, StarkProof};
-pub use processor::{AdviceInputs, StackInputs};
-use processor::{ExecutionError, ExecutionTrace, Process, VmStateIterator};
-use proptest::prelude::*;
-use stdlib::StdLibrary;
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+// IMPORTS
+// ================================================================================================
+#[cfg(feature = "std")]
+use proptest::prelude::{Arbitrary, Strategy};
+
+use vm_core::utils::{collections::Vec, string::String};
+
+// EXPORTS
+// ================================================================================================
+
+pub use assembly::{Library, MaslLibrary};
+pub use processor::{
+    AdviceInputs, AdviceProvider, ExecutionError, ExecutionTrace, Process, StackInputs,
+    VmStateIterator,
+};
+pub use prover::{MemAdviceProvider, ProofOptions};
+pub use test_case::test_case;
+pub use verifier::ProgramInfo;
 pub use vm_core::{
-    crypto::merkle::MerkleStore, stack::STACK_TOP_SIZE, Felt, FieldElement, Program, StackOutputs,
+    stack::STACK_TOP_SIZE,
+    utils::{collections, group_slice_elements, group_vector_elements, IntoBytes, ToElements},
+    Felt, FieldElement, Program, StarkField, Word, ONE, WORD_SIZE, ZERO,
 };
 
+pub mod math {
+    pub use winter_prover::math::{fft, polynom};
+}
+
+pub mod serde {
+    pub use vm_core::utils::{Deserializable, DeserializationError, Serializable, SliceReader};
+}
+
 pub mod crypto;
 
+mod test_builders;
+pub use test_builders::*;
+
+#[cfg(feature = "std")]
+pub use proptest;
+
+#[cfg(feature = "std")]
+pub use rand_utils as rand;
```

Comment:
These two won't be available in no_std environment, is this why CI tests are failing ?

Reply:
Now that I look at CI log, it's different.

Reply:
Yeah - the issue is that we have a check for the whole thing in `no_std`. I changed this to make them unavailable only for `wasm` targets. Not, but given our current CI setup - not sure if there is a better way.

Reply:
I see.

> Not, but given our current CI setup - not sure if there is a better way. 

I don't know though.